turbo.json

```diff
@@ -4,15 +4,23 @@
   "pipeline": {
     "build": {
       "dependsOn": ["^build"],
-      "outputs": [".next/**", "artifacts/**", "typechain-types/**"]
+      "outputs": [".next/**", "deployments/**", "artifacts/**", "typechain-types/**"]
     },
-    "@yieldgate/frontend#build": {
+    "@yieldgate/frontend-yg#dev": {
       "dependsOn": ["@yieldgate/contracts#build"],
-      "outputs": [".next/**", "artifacts/**", "typechain-types/**"]
+      "outputs": [".next/**", "deployments/**", "artifacts/**", "typechain-types/**"]
     },
-    "@yieldgate/frontend#dev": {
+    "@yieldgate/frontend-sfe#dev": {
       "dependsOn": ["@yieldgate/contracts#build"],
-      "outputs": [".next/**", "artifacts/**", "typechain-types/**"]
+      "outputs": [".next/**", "deployments/**", "artifacts/**", "typechain-types/**"]
+    },
+    "@yieldgate/frontend-yg#build": {
+      "dependsOn": ["@yieldgate/contracts#build"],
+      "outputs": [".next/**", "deployments/**", "artifacts/**", "typechain-types/**"]
+    },
+    "@yieldgate/frontend-sfe#build": {
+      "dependsOn": ["@yieldgate/contracts#build"],
+      "outputs": [".next/**", "deployments/**", "artifacts/**", "typechain-types/**"]
```

Comment:
Didn't you remove `deployments` from the frontend packages and moved them into `contracts`?

Reply:
Yes I did, I assumed `outputs` affects not only the output directory of the task itself but also its depending tasks (i.e. `@yieldgate/contracts#build`). I'm not 100% sure how it's supposed to work even after reading the docs (https://turborepo.org/docs/reference/configuration#outputs). Btw `artifacts` and `typechain-types` are also contract-only directories and we have them in there since when we started using turborepo.

So would you think something like that works better:

```json
    "build": {
      "dependsOn": ["^build"],
      "outputs": [".next/**", "deployments/**", "artifacts/**", "typechain-types/**"]
    },
    // added:
    "@yieldgate/contracts#build": {
      "dependsOn": [],
      "outputs": ["deployments/**", "artifacts/**", "typechain-types/**"]
    },
    // modified:
    "@yieldgate/frontend-yg#dev": {
      "dependsOn": ["@yieldgate/contracts#build"],
      "outputs": [".next/**"]
    },
    "@yieldgate/frontend-sfe#dev": {
      "dependsOn": ["@yieldgate/contracts#build"],
      "outputs": [".next/**"]
    },
    "@yieldgate/frontend-yg#build": {
      "dependsOn": ["@yieldgate/contracts#build"],
      "outputs": [".next/**"]
    },
    "@yieldgate/frontend-sfe#build": {
      "dependsOn": ["@yieldgate/contracts#build"],
      "outputs": [".next/**"]
    },
```

Reply:
I just ran `turbo build` with this simplified `pipeline` definition and it properly first build the `contracts` and only then the two frontend packages in parallel. So the `^build` dependency seems to properly resolve the dependency of the two frontend packages on the contracts already.

Reply:
Thanks for your investigation. I think you are right, we don't need individual definitions. With one further change (as we don't want frontends to be built during the `dev` task) all tasks seem to work fine. And as `outputs` are not inherited I defined them there again. What do you think?

```json
    "build": {
      "dependsOn": ["^build"],
      "outputs": [".next/**", "deployments/**", "artifacts/**", "typechain-types/**"]
    },
    "dev": {
      "dependsOn": ["@yieldgate/contracts#build"],
      "outputs": [".next/**", "deployments/**", "artifacts/**", "typechain-types/**"]
    },
    "lint": {
      "outputs": []
    },
    "//#lint": {
      "outputs": []
    },
    "lint:fix": {},
    "//#lint:fix": {}
```

Btw I just discovered the turbo CLI has this nice `--dry` flag where it just outputs the tasks that will be run and their according configs.

Reply:
Simplified `turbo.json` in https://github.com/yieldgate/yieldgate/pull/63/commits/f4954fa80dce189aa3664f0439664329b002c444.